owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ReceiveExternalFilesAdapter.java

```diff
@@ -39,32 +40,31 @@
 import com.owncloud.android.datamodel.ThumbnailsCacheManager;
 import com.owncloud.android.datamodel.ThumbnailsCacheManager.AsyncThumbnailDrawable;
 import com.owncloud.android.db.PreferenceManager;
+import com.owncloud.android.extensions.AdapterExtKt;
 import com.owncloud.android.utils.DisplayUtils;
 import com.owncloud.android.utils.FileStorageUtils;
 import com.owncloud.android.utils.MimetypeIconUtil;
 import com.owncloud.android.utils.PreferenceUtils;
 
-import java.util.ArrayList;
-import java.util.List;
 import java.util.Vector;
 
 public class ReceiveExternalFilesAdapter extends BaseAdapter implements ListAdapter {
 
-    private List<OCFile> mImmutableFilesList = new ArrayList<>();
-    private List<OCFile> mFiles;
+    private Vector<OCFile> mImmutableFilesList;
+    private Vector<OCFile> mFiles;
     private Context mContext;
     private Account mAccount;
     private FileDataStorageManager mStorageManager;
     private LayoutInflater mInflater;
     private OnSearchQueryUpdateListener mOnSearchQueryUpdateListener;
 
     public ReceiveExternalFilesAdapter(Context context,
-                                       List<OCFile> files,
+                                       Vector<OCFile> files,
                                        FileDataStorageManager storageManager,
                                        Account account
     ) {
-        mImmutableFilesList.addAll(files);
         mFiles = files;
+        mImmutableFilesList = (Vector<OCFile>) mFiles.clone();
```

Comment:
I'm not sure about clone implementation. Can you keep the previous behaviour?

Reply:
This is the existing implementation for other adapters. Check the FileListListAdapter in line 474. But if you prefer the previous implementation i will change it.

Reply:
In this case, `mImmutableFilesList` is a clone of `mFiles`.
In FileListListAdapter, `mFiles` is a clone of the immutable one.

I don't use Vector so often so i'm not 100% confident about `clone` uses. But since it seems to work fine, you can keep it this way. Check the other comment and rebase with latest master 👍